### PR TITLE
HAAR-862: Added Create token by email type and user name

### PR DESCRIPTION
**Pull request by ravmoj**
This is one of the step in Auth split process for endpoint amend email id of user

### CODE REVIEW

File: src/test/kotlin/uk/gov/justice/digital/hmpps/oauth2server/verify/TokenServiceTest.kt [hunk 1]
```diff
@@ -267,4 +269,26 @@ class TokenServiceTest {
       isNull()
     )
   }
+  @Nested
+  inner class CreateTokenByEmailType {
```
Comment (AlTurner-MOJ) [inline src/test/kotlin/uk/gov/justice/digital/hmpps/oauth2server/verify/TokenServiceTest.kt:273]: These tests should verify the different parameter passed to the createToken call when the email type is PRIMARY or SECONDARY

File: src/test/kotlin/uk/gov/justice/digital/hmpps/oauth2server/resource/api/TokenControllerIntTest.kt [hunk 2]
```diff
@@ -19,95 +21,181 @@ class TokenControllerIntTest : IntegrationTest() {
...
+        .expectStatus().isBadRequest
+        .expectBody()
+        .json(
+          """
+        {"status":400,"error":"Bad Request","path":"/auth/api/token/email-type"} 
+          """.trimIndent()
+        )
+    }
+
+    /*
```
Comment (AlTurner-MOJ) [inline src/test/kotlin/uk/gov/justice/digital/hmpps/oauth2server/resource/api/TokenControllerIntTest.kt:191]: Remove this if not required

File: src/test/kotlin/uk/gov/justice/digital/hmpps/oauth2server/verify/TokenServiceTest.kt (hunk 1 above)
Comment (ravmoj) [reply src/test/kotlin/uk/gov/justice/digital/hmpps/oauth2server/verify/TokenServiceTest.kt:273]: removed duplicate test

File: src/test/kotlin/uk/gov/justice/digital/hmpps/oauth2server/resource/api/TokenControllerIntTest.kt (hunk 2 above)
Comment (ravmoj) [reply src/test/kotlin/uk/gov/justice/digital/hmpps/oauth2server/resource/api/TokenControllerIntTest.kt]: removed

**Review by AlTurner-MOJ — APPROVED** [review]
lgtm

**Pull request merged**